Sources/Sentry/SentryNSDataSwizzling.m

```diff
@@ -0,0 +1,104 @@
+#import "SentryNSDataSwizzling.h"
+#import "SentryNSDataTracker.h"
+#import "SentrySwizzle.h"
+#import <SentryLog.h>
+#import <objc/runtime.h>
+
+@implementation SentryNSDataSwizzling
```

Comment:
`h`: What are our plans to ensure we are not breaking the functionality of `NSData`?

Reply:
This is essential. 
We should create as many tests as possible.